### PR TITLE
Update cute_files.h to upstream f04e37c5a8ba1c7ce385b0f38b584bec1056e831

### LINKED ISSUE #57: ../src/external/cute_files.h:427: int cf_dir_open(cf_dir_t*, const char*): Assertion `0' failed on ClangBuildAnalyzer --stop
Recently, when trying to use ClangBuildAnalyzer to analyze a firefox build, I got the following error and assertion failure on `ClangBuildAnalyzer --stop`:
```
ERROR: Failed to open directory (obj-x86_64-pc-linux-gnu-trace/dist/bin/chrome/en-US/locale/en-US/mozapps/downloads/settingsChange.dtd): No such file or directory.
ClangBuildAnalyzer: ../src/external/cute_files.h:427: int cf_dir_open(cf_dir_t*, const char*): Assertion `0' failed.
```

`obj-x86_64-pc-linux-gnu-trace/dist/bin/chrome/en-US/locale/en-US/mozapps/downloads/settingsChange.dtd` exists, but is a symbolic link to a file, not a directory:
```
[simon@sigibln fuzzy]$ stat obj-x86_64-pc-linux-gnu-trace/dist/bin/chrome/en-US/locale/en-US/mozapps/downloads/settingsChange.dtd
  File: obj-x86_64-pc-linux-gnu-trace/dist/bin/chrome/en-US/locale/en-US/mozapps/downloads/settingsChange.dtd -> /home/simon/work/fuzzy/toolkit/locales/en-US/chrome/mozapps/downloads/settingsChange.dtd
  Size: 88              Blocks: 8          IO Block: 4096   symbolic link
Device: fd02h/64770d    Inode: 34560445    Links: 1
Access: (0777/lrwxrwxrwx)  Uid: ( 1000/   simon)   Gid: ( 1000/   simon)
Context: unconfined_u:object_r:user_home_t:s0
Access: 2020-10-21 18:42:33.948517287 +0200
Modify: 2020-10-14 18:48:03.012265746 +0200
Change: 2020-10-14 18:48:03.012265746 +0200
 Birth: 2020-10-14 18:48:03.012265746 +0200
[simon@sigibln fuzzy]$ stat /home/simon/work/fuzzy/toolkit/locales/en-US/chrome/mozapps/downloads/settingsChange.dtd
  File: /home/simon/work/fuzzy/toolkit/locales/en-US/chrome/mozapps/downloads/settingsChange.dtd
  Size: 0               Blocks: 0          IO Block: 4096   regular empty file
Device: fd02h/64770d    Inode: 49026953    Links: 1
Access: (0664/-rw-rw-r--)  Uid: ( 1000/   simon)   Gid: ( 1000/   simon)
Context: unconfined_u:object_r:user_home_t:s0
Access: 2020-10-22 09:43:44.048022446 +0200
Modify: 2020-10-22 09:43:34.870015064 +0200
Change: 2020-10-22 09:43:34.870015064 +0200
 Birth: 2020-10-22 09:43:29.366010639 +0200
```

### PR DESCRIPTION
**Pull request by sigiesec**
Fixes #57

### CODE REVIEW

File: src/external/cute_files.h [hunk 1]
```diff
@@ -73,7 +73,7 @@
 
 #define CUTE_FILES_MAX_PATH 1024
 #define CUTE_FILES_MAX_FILENAME 256
-#define CUTE_FILES_MAX_EXT 64
+#define CUTE_FILES_MAX_EXT 32
```
Comment (aras-p) [inline src/external/cute_files.h]: Maybe revert this back to 64, IIRC someone had builds with "very very long" extensions and bumped it up to 64

**Review by aras-p — CHANGES_REQUESTED** [review]
Maybe max extension length should be left as is (64), otherwise LGTM

File: src/external/cute_files.h (hunk 1 above)
Comment (sigiesec) [reply src/external/cute_files.h]: Ok

**Comment by sigiesec** [comment]
@aras-p I reverted that change as you requested, can this be merged now?

**Review by aras-p — APPROVED** [review]

**Pull request merged**